src/pretix/plugins/stripe/views.py

```diff
@@ -467,3 +474,37 @@ def _redirect_to_order(self):
             'order': self.order.code,
             'secret': self.order.secret
         }) + ('?paid=yes' if self.order.status == Order.STATUS_PAID else ''))
+
+
+@method_decorator(xframe_options_exempt, 'dispatch')
+class ScaView(StripeOrderView, View):
+    def get(self, request, *args, **kwargs):
+        prov = self.pprov
+        prov._init_api()
+
+        payment_info = json.loads(self.payment.info)
```

Comment:
Could we start this view off by checking whether ``self.payment.state`` is already in a "final" state like ``PAYMENT_STATE_CONFIRMED``, ``PAYMENT_STATE_CANCELED``, or ``PAYMENT_STATE_FAILED`` and then redirect directly to the order without anything else? For the confirmed case, this would increase performance, for the other cases, it would prevent irregular behaviour.